features/configuration/fail_if_no_examples.feature

```diff
@@ -0,0 +1,106 @@
+   when there are 0 examples.
+
+  Scenario: A failing spec example and `fail_if_no_examples` set to true
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = true }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "fails" do
+          fail
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 1
+
+  Scenario: A failing spec example and `fail_if_no_examples` set to false
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = false }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "fails" do
+          fail
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 1
+
+  Scenario: A passing spec example and `fail_if_no_examples` set to true
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = true }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "succeeds" do
+          true
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 0
+
+  Scenario: A passing spec example and `fail_if_no_examples` set to false
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = false }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+        it "succeeds" do
+          true
+        end
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 0
+
+  Scenario: No examples and `fail_if_no_examples` set to true
+    Given a file named "spec/spec_helper.rb" with:
+      """ruby
+      RSpec.configure { |c| c.fail_if_no_examples = true }
+      """
+    Given a file named "spec/example_spec.rb" with:
+      """ruby
+      require 'spec_helper'
+
+      RSpec.describe "something" do
+      end
+      """
+    When I run `rspec spec/example_spec.rb`
+    Then the exit status should be 1
+
+  Scenario: No examples and `fail_if_no_examples` set to false
+   Given a file named "spec/spec_helper.rb" with:
+     """ruby
+     RSpec.configure { |c| c.fail_if_no_examples = false }
+     """
+   Given a file named "spec/example_spec.rb" with:
+     """ruby
+     require 'spec_helper'
+
+     RSpec.describe "something" do
+     end
+     """
+   When I run `rspec spec/example_spec.rb`
+   Then the exit status should be 0
```

Comment:
There are a few things about this cuke that can be improved, IMO:

* It's super low level and detailed, and doesn't function very well as documentation.  (Although, it is similar to some of our other cukes, but we have many that need to be cleaned up).
* Having test coverage only in a cuke means `rspec spec` (which is the main thing we use locally) can't be used to tell us if this feature is still working.

Putting these together, here's what I'd like to see:

* This feature should be covered by specs, and not just the cukes.  You can add an integration spec to `spec/integration` if you want to test it in an integration style.
* The cuke here should really be focused on documentation, and not on exhaustively showing every edge case.  Here's what I'm thinking it needs:
  * A narrative fleshing out the reason behind this feature and how to use it.  It should include a note about how if you use this you probably want to add a `--require` option to `.rspec` to ensure the option is always set even if no spec files are loaded by the command.
  * A background setting up a common context for the entire file:
    * A `spec_helper` file that sets the flag.
    * A `.rspec` file that has `--require spec_helper` to demonstrate how we recommend users set things up
    * A spec file not matching the normal spec file pattern (e.g. `some.spec` instead of `some_spec.rb`).
  * In one scenario, run `rspec` (with no options) and assert `Then it should fail with "0 examples, 0 failures"`.
  * I a 2nd scenario, run `rspec --pattern spec/**/*.spec` so the spec file is matched, and assert `Then it should pass with 1 example, 0 failures`

That way, there are 2 short scenarios that are very easy to read and understand, a clear explanation of the reason for the feature and how to set it up, and not much else to distract from that core documentation purpose.

(Also, this new file needs to be added to `features/.nav`).

Thoughts?